test/smoke/src/helpers/setupEnvironmentHelper.ts

```diff
@@ -91,6 +90,46 @@ export function prepareReactNativeApplication(workspaceFilePath: string, resourc
     fs.writeFileSync(path.join(vsCodeConfigPath, "launch.json"), fs.readFileSync(launchConfigFile));
 }
 
+export function installExtensionVSIX(extensionDir: string, testVSCodeExecutablePath: string, resourcesPath: string, isInsiders: boolean) {
+    let args: string[] = [];
+    args.push(`--extensions-dir=${extensionDir}`);
+    let dirFiles = fs.readdirSync(resourcesPath);
+    let extensionFile = dirFiles.find((elem) => {
+        return elem.match(/.*\.(vsix)/);
+    });
+    if (!extensionFile) throw new Error(`React Native extension .vsix is not found in ${resourcesPath}`);
+
+    extensionFile = path.join(resourcesPath, extensionFile);
+    args.push(`--install-extension=${extensionFile}`);
+    let codeExecutableScript = isInsiders ? "code-insiders" : "code";
+    testVSCodeExecutablePath = path.join(testVSCodeExecutablePath, codeExecutableScript);
+    if (process.platform === "win32") {
+        testVSCodeExecutablePath += ".cmd";
+    }
+    console.log(`*** Installing ${extensionFile} into ${extensionDir} with ${testVSCodeExecutablePath} executable`);
+    cp.spawnSync(testVSCodeExecutablePath, args, {stdio: "inherit"});
+    console.log(`*** Deleting ${extensionFile} after installation`);
+    rimraf.sync(extensionFile);
+}
+
+export function runAndroidEmulator() {
+    if (!process.env.ANDROID_EMULATOR) throw new Error("Environment variable 'ANDROID_EMULATOR' is not set. Exiting...");
+    terminateAndroidEmulator();
+    console.log(`*** Executing Android emulator with 'emulator -avd ${process.env.ANDROID_EMULATOR}' command...`);
+    cp.spawn("emulator", ["-avd", process.env.ANDROID_EMULATOR || "", "-wipe-data"], {stdio: "inherit"});
+
+}
+
+export function terminateAndroidEmulator() {
+    let devices = cp.execSync("adb devices", {stdio: "inherit"}).toString();
+    console.log("*** Checking for running emulators...");
+    if (devices !== "List of devices attached\r\n\r\n") {
+        // Check if we already have a running emulator, and terminate it if it so
+        console.log("Terminating Android 'emulator-5554'...");
+        cp.execSync("adb -s emulator-5554 emu kill", {stdio: "inherit"});
```

Comment:
Are you sure, that emulator would be always named as `emulator-5554`?

Reply:
I added `port` parameter to the emulator execute command. So it name would be `emulator-` + `port`.